src/sdk/pynni/nni/compression/tensorflow/builtin_pruners.py

```diff
@@ -94,3 +94,83 @@ def update_epoch(self, epoch, sess):
         sess.run(tf.assign(self.now_epoch, int(epoch)))
         for k in self.if_init_list:
             self.if_init_list[k] = True
+
+class FPGMPruner(Pruner):
+    """A filter pruner via geometric median.
+    "Filter Pruning via Geometric Median for Deep Convolutional Neural Networks Acceleration", 
+    https://arxiv.org/pdf/1811.00250.pdf
+    """
+
+    def __init__(self, config_list):
+        """
+        config_list: supported keys:
+            - pruning_rate: percentage of convolutional filters to be pruned.
+        """
+        super().__init__(config_list)
+        self.mask_list = {}
+        self.assign_handler = []
+
+    def calc_mask(self, conv_kernel_weight, config, op, op_type, op_name, **kwargs):
+        """supports Conv1d, Conv2d, Conv3d
+        filter dimensions for Conv1D:
```

Comment:
Doc string format?

Reply:
tensorflow version will be reimplemented later.

Reply:
Doc string updated.